### DISCUSSION TITLE
montage: command not found

### DISCUSSION THREAD

**Discussion by Leopy96** [ImageMagick/ImageMagick#1969] [General] [answered]
Hi everyone. Probably a stupid question that can be solved in a couple of minutes, but in 1h of research I have found no clue on what to do.
I attach a picture explaining the issue; basically, I'm trying to run on Windows [this script](https://opentopomap.org/files/montage.sh) provided by OpenTopoMap that downloads a bunch of map tiles in png format and then stitches them together to form a complete map image.
The script runs up to the point where Imagemagick is called, but stops.
I have used the Windows installer ("ImageMagick-7.0.10-10-Q16-x64-dll.exe", at the end of the official [download page](https://imagemagick.org/script/download.php)) , and I'm able to run correctly some "independent" commands from the cmd prompt (such as "magick logo: logo.gif"), but I cannot seem to find a way to run the montage command from the script given.
During installation I have checked all the checkboxes needed for external packages and environment variables (although I am not sure this last one has been correctly added...).
The only hint I have found so far is [this post](https://github.com/astropy/montage-wrapper/issues/63) , but I have no idea on what it even means to "make" a file. I can't even find the path that they talk about...

Thanks a lot! I'm quite a noob about these things... having always written only C++ programs...

[ Edit # 1: I forgot to mention I have also installed Windows Subsystem for Linux (I followed the first point in [this guide](https://www.thewindowsclub.com/how-to-run-sh-or-shell-script-file-in-windows-10)) ]

![prompt](https://user-images.githubusercontent.com/64611902/80755073-72446f00-8b30-11ea-8c8c-76734f801236.png)

**Comment by fmw42** [comment, marked as answer]
That is not a Windows script. It is a bash Unix script and does not run on Windows as is. You would need to run it under Windows 10 Unix in a bash shell window, not a CMD window. Also for ImageMagick 7 that you are using, montage needs to become magick montage.

**Reply by Leopy96** [reply]
Yes, I forgot to mention I have also installed Windows Subsystem for Linux (I followed the first point in this [guide](https://www.thewindowsclub.com/how-to-run-sh-or-shell-script-file-in-windows-10)) ; that's the reason the script is able to work up to a point (I also had to format the text specifically with dos2unix in order to remove the unnecessary '\r').
I have tried adding the word "magick" before "montage" , but I get a very similar error: "magick: command not found" , whereas before it was "montage: command not found".

Does this sound correct to you? Or I have made some other silly mistake in the process?
Thanks, in the meantime.

**Reply by fmw42** [reply]
Does your Window 10 Unix have ImageMagick installed or did you install ImageMagick in Window DOS?  Type

magick -version 

in your Unix system?  What do you get? If an error message, then you likely need to install ImageMagick Unix version in your Unix Windows system.

**Reply by Leopy96** [reply]
Here's what I get:
"
C:\WINDOWS\system32>magick -version
Version: ImageMagick 7.0.10-10 Q16 x64 2020-04-28 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2018 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Visual C++: 180040629
Features: Cipher DPC Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zlib
"

**Reply by fmw42** [reply]
But you are running that command from a CMD window and not in the Unix window, if I understand. Though I am not a Windows user. Can or did you run the same from a bash shell window in your Unix subsystem?

**Reply by Leopy96** [reply]
I'm not sure if I can fully understand your question; I've just remembered that I also installed Kali linux app from the MS store (you can tell I have tried many ways, not knowing where I was going hahahah), and if I try to run the bash file there I get a similar error, but this time if I ask for magick version I get an error.
Am I close to the answer you were asking for?
![Immagine](https://user-images.githubusercontent.com/64611902/80761567-17644500-8b3b-11ea-82b2-49c4f141edb4.jpg)

**Reply by fmw42** [reply]
When you use Windows 10, you have two systems: DOS and Unix.  You have been using and likely installed Imagemgick only in the DOS Windows system and not in the Unix subsystem. I am not a Windows user and so cannot help much. But you need to install a Unix version of ImageMagick into your Unix subsystem. Then launch a bash terminal window (not a CMD window) from within your Unix subsystem to be able to run that script.

**Reply by Leopy96** [reply]
That actually makes a lot of sense 👍 I will try tomorrow, in the meantime thanks a lot!

**Reply by Stevieb36** [reply]
Hi,
I'm new to ImageMagick and I have never coded before, but I want to stitch together some map tiles just like you did and would appreciate some help. I have installed the Windows version of ImageMagick and I'm just starting to (struggling with learning what to do) stitching together my map tiles.  I have an 8x8 matrix and I have labeled my files as follows:
A1 A2 A3 A4...
B1 B2 B3 B4...
C1 C2 C3 C3...
etc.

It sounds like you may have already had success stitching together the tiles. Any help with the command line prompts would be greatly appreciated or could point me to a good tutorial.

**Reply by snibgo** [reply]
@Stevieb36: Your question has no relevance to the original question. Please start a new thread.

**Comment by snibgo** [comment]
You can either:

1. Run the "magick" command from the Windows CMD prompt.

2. Re-write the bash script as a BAT script.

3. Learn how to run bash on your system.

**Reply by Leopy96** [reply]
Thanks! I will try a couple of those tomorrow and see which one is the best to implement for me!
